### PR TITLE
[niconico] add support for series

### PR DESCRIPTION
**Pull request by aaaaaaaaaaaaaaaaa65**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information
nicovideo.jp has an infrequently used feature called a "series" [example: https://www.nicovideo.jp/series/8253](https://www.nicovideo.jp/series/8253) that creators can use to make a playlist for some subset of their uploads. All series can be represented as normal (mylist) playlists but not all are.

### CODE REVIEW

File: youtube_dl/extractor/niconico.py [hunk 1]
```diff
@@ -468,3 +469,29 @@ def _real_extract(self, url):
...
+    _TEST = {
+        'url': 'https://www.nicovideo.jp/series/8253',
+        'info_dict': {
+            'id': '8253',
+            'title': '弦巻マキと結月ゆかりの未確認ゲーム日和',
+        },
+        'playlist_mincount': 49,
+    }
+    def _real_extract(self, url):
+        series_id = url.split('/')[-1]
```
Comment (dstftw) [inline youtube_dl/extractor/niconico.py]: `_match_id`.

File: youtube_dl/extractor/niconico.py [hunk 2]
```diff
@@ -468,3 +469,29 @@ def _real_extract(self, url):
...
+        webpage = self._download_webpage(url, series_id)
+        entries = re.findall(r'<a href=["\'](\/watch\/[^"\']*)', webpage)
```
Comment (dstftw) [inline youtube_dl/extractor/niconico.py]: Inline.

File: youtube_dl/extractor/niconico.py [hunk 3]
```diff
@@ -468,3 +469,29 @@ def _real_extract(self, url):
...
+        entries = [{
+            '_type': 'url',
+            'ie_key': NiconicoIE.ie_key(),
+            'url': ('https://www.nicovideo.jp%s' % entry),
```
Comment (dstftw) [inline youtube_dl/extractor/niconico.py]: No parens.

Comment (dstftw) [inline youtube_dl/extractor/niconico.py]: url_join.

File: youtube_dl/extractor/niconico.py [hunk 4]
```diff
@@ -468,3 +469,29 @@ def _real_extract(self, url):
...
+        } for entry in entries]
+        return {
+            '_type': 'playlist',
+            'id': series_id,
+            'entries': entries,
+            'title': self._search_regex(r'bodyTitle">(.*?)</div>', webpage, 'title'),
```
Comment (dstftw) [inline youtube_dl/extractor/niconico.py]: Must not be fatal.

File: youtube_dl/extractor/niconico.py (hunk 2 above)
Comment (dstftw) [inline youtube_dl/extractor/niconico.py]: No escapes for slash.

Comment (dstftw) [inline youtube_dl/extractor/niconico.py]: Do not capture empty entries.

**Comment by gj-thompson** [comment]
Fixes are no longer pending in my estimation. Can I get a merge or some clarification on the issues raised earlier such that I'd need to change something?

**Pull request closed without merging**